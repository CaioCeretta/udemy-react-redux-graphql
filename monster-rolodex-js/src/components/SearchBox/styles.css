.search-box {
  -webkit-appearance: none;
  border: none;
  outline: none;
  padding: 10px;
  width: 150px;
  line-height: 30px;
  margin-bottom: 30px;
}

/* .card-list {
  background: red; /*= This is to show that when we import the css it is aplicable to the whole site 
} 
*/
