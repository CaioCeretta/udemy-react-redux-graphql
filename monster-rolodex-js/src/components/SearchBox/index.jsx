import { Component } from "react";
import './styles.css'

export class SearchBox extends Component {

  render () {
    return (
      <input
        type="text"
        className={`search-box ${this.props.className}`}
        type="search"
        placeholder={this.props.place}
        onChange={this.props.onChangeHandler}
      />
    )
  }

}