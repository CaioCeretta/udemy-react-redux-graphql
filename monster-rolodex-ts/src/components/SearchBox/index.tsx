import { ChangeEvent } from 'react';
import './styles.css';

interface ISearchBoxProps {
  placeholder?: string
  className: string;
  onChangeHandler: (e: ChangeEvent<HTMLInputElement>) => void;

}


function SearchBox ({placeholder, onChangeHandler, className}: ISearchBoxProps) {

    return (
      <input 
        className={`search-box ${className}`}
        type="search"
        placeholder={placeholder ? placeholder : ''}
        onChange={onChangeHandler}
      />
    )

}

export { SearchBox };
