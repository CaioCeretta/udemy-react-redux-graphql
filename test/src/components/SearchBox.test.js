import { shallow } from "enzyme"
import SearchBox from "./SearchBox"

it('Expects the Search Box to be rendered', () => {
  const mockSearchField = 'jorge'


  expect(shallow(<SearchBox searchfield={mockSearchField} searchChange={() => {}}/>)).toMatchSnapshot()
})