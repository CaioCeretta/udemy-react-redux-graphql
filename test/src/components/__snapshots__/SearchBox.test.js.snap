// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expects the Search Box to be rendered 1`] = `
<div
  className="pa2"
>
  <input
    aria-label="Search"
    className="pa3 ba b--green bg-lightest-blue"
    onChange={[Function]}
    placeholder="search robots"
    type="search"
  />
</div>
`;
